[2025-06-11T15:04:46.036+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: multi_region_transform_staging_v2.process_region_us.process_categories_us scheduled__2025-06-10T00:00:00+00:00 [queued]>
[2025-06-11T15:04:46.053+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: multi_region_transform_staging_v2.process_region_us.process_categories_us scheduled__2025-06-10T00:00:00+00:00 [queued]>
[2025-06-11T15:04:46.054+0000] {taskinstance.py:2170} INFO - Starting attempt 3 of 2
[2025-06-11T15:04:46.087+0000] {taskinstance.py:2191} INFO - Executing <Task(_PythonDecoratedOperator): process_region_us.process_categories_us> on 2025-06-10 00:00:00+00:00
[2025-06-11T15:04:46.104+0000] {standard_task_runner.py:60} INFO - Started process 318 to run task
[2025-06-11T15:04:46.109+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'multi_region_transform_staging_v2', 'process_region_us.process_categories_us', 'scheduled__2025-06-10T00:00:00+00:00', '--job-id', '77', '--raw', '--subdir', 'DAGS_FOLDER/master_dag.py', '--cfg-path', '/tmp/tmp3jz9i0_7']
[2025-06-11T15:04:46.112+0000] {standard_task_runner.py:88} INFO - Job 77: Subtask process_region_us.process_categories_us
[2025-06-11T15:04:46.237+0000] {task_command.py:423} INFO - Running <TaskInstance: multi_region_transform_staging_v2.process_region_us.process_categories_us scheduled__2025-06-10T00:00:00+00:00 [running]> on host 94d2ab00ac05
[2025-06-11T15:04:46.455+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='data-team' AIRFLOW_CTX_DAG_ID='multi_region_transform_staging_v2' AIRFLOW_CTX_TASK_ID='process_region_us.process_categories_us' AIRFLOW_CTX_EXECUTION_DATE='2025-06-10T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-06-10T00:00:00+00:00'
[2025-06-11T15:05:19.680+0000] {logging_mixin.py:188} INFO - Debugggg::::Vector(spark://94d2ab00ac05:39415/jars/snowflake-jdbc-3.13.17.jar, spark://94d2ab00ac05:39415/jars/spark-snowflake_2.12-2.16.0-spark_3.2.jar)
[2025-06-11T15:05:19.716+0000] {logging_mixin.py:188} INFO - Processing categories for region us
[2025-06-11T15:05:19.798+0000] {logging_mixin.py:188} INFO - /opt/airflow/data/raw/region=us/table=categories/load_date=2025-06-11
[2025-06-11T15:08:10.193+0000] {logging_mixin.py:188} INFO - Error processing categories for us: transform_categories() takes 2 positional arguments but 3 were given
[2025-06-11T15:08:12.229+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 241, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/master_dag.py", line 242, in process_table
    df_transformed = transform_func(df_raw, region,exchange_rates)
TypeError: transform_categories() takes 2 positional arguments but 3 were given
[2025-06-11T15:08:13.315+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=multi_region_transform_staging_v2, task_id=process_region_us.process_categories_us, execution_date=20250610T000000, start_date=20250611T150446, end_date=20250611T150813
[2025-06-11T15:08:14.751+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 77 for task process_region_us.process_categories_us (transform_categories() takes 2 positional arguments but 3 were given; 318)
[2025-06-11T15:08:14.827+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-06-11T15:08:17.543+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
